build-conventions-secant/src/main/kotlin/secant.publish-conventions.gradle.kts

```
@@ -0,0 +1,395 @@
+import com.google.api.client.util.ExponentialBackOff
+import com.google.api.services.androidpublisher.AndroidPublisher
+import com.google.api.services.androidpublisher.AndroidPublisherRequestInitializer
+import com.google.api.services.androidpublisher.AndroidPublisherScopes
+import com.google.api.services.androidpublisher.model.AppEdit
+import com.google.api.services.androidpublisher.model.Bundle
+import com.google.api.services.androidpublisher.model.Track
+import com.google.api.services.androidpublisher.model.TrackRelease
+import com.google.auth.http.HttpCredentialsAdapter
+import com.google.auth.oauth2.GoogleCredentials
+import org.apache.http.auth.AuthScope
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.client.ProxyAuthenticationStrategy
+import java.io.FileInputStream
+import java.io.IOException
+import java.security.GeneralSecurityException
+import java.security.KeyStore
+
+@CacheableTask
+abstract class PublishToGooglePlay @Inject constructor(
+    private val gpServiceAccountKey: String,
+    private val gpPublisherApiKey: String,
+    private val track: String,
+    private val status: String
+) : DefaultTask() {
+
+    // Note that we need to have all the necessary custom task properties part of the task (i.e. no external
+    // dependencies allowed) to avoid:
+    // PublishToGooglePlay is a non-static inner class.
+
+    init {
+        description = "Publish universal Zcash wallet apk to Google Play release channel."    // $NON-NLS-1$
+        group = "publishing"    // $NON-NLS-1$
+    }
+
+    private fun log(message: String) {
+        println("${PublishToGooglePlay::class.java.name}: $message")
+    }
+
+    // Global instance of the JSON factory
+    private val jsonFactory: JsonFactory by lazy {
+        GsonFactory.getDefaultInstance()
+    }
+
+    // Global instance of the HTTP transport
+    @get:Throws(GeneralSecurityException::class, IOException::class)
+    private val trustedTransport: HttpTransport by lazy {
+        buildTransport()
+    }
+
+    /**
+     * Prepares a new trusted [HttpTransport] object to authorize [AndroidPublisher] on Google Play Publish API.
+     */
+    private fun buildTransport(): HttpTransport {
+        val trustStore: String? = System.getProperty("javax.net.ssl.trustStore", null)
+        val trustStorePassword: String? =
+            System.getProperty("javax.net.ssl.trustStorePassword", null)
+
+        return if (trustStore == null) {
+            createHttpTransport()
+        } else {
+            val ks = KeyStore.getInstance(KeyStore.getDefaultType())
+            FileInputStream(trustStore).use { fis ->
+                ks.load(fis, trustStorePassword?.toCharArray())
+            }
+            NetHttpTransport.Builder().trustCertificates(ks).build()
+        }
+    }
+
+    private fun createHttpTransport(): HttpTransport {
+        val protocols = arrayOf("http", "https")
```

```suggestion
        val protocols = arrayOf("https", "http")
```
to prefer https if both sets of properties are set.